test/docker-e2e/run_upgrade_test.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
```

Comment:
I know this PR is in draft right now, but a separate bash script to run this test is overkill IMO. I guess it depends on what the expected usage of this tool is, but was thinking we can keep things simple with just run things with the existing make e2e-test target or a slight variation and specify some before and after tags. No need for maximum flexibility in the first iteration.